### PR TITLE
Use CapacityHeatmapSceneBuilder in ResourceLoadRenderer

### PR DESCRIPTION
**Pull request by DmKazakov**
The first commit just copies code from GP cloud

### CODE REVIEW

**Comment by dbarashev** [comment]
I suggest moving CapacityHeatmapSceneBuilder into biz.ganttproject.core

File: biz.ganttproject.core/src/biz/ganttproject/core/chart/scene/CapacityHeatmapSceneBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  private fun createLoadText(rect: Canvas.Rectangle, load: Float) {
+    val loadLabel = canvas.createText(rect.middleX, rect.middleY - input.yCanvasOffset, "")
+    loadLabel.setSelector { textLengthCalculator ->
+      val loadInt = load.roundToInt()
+      val loadStr = "$loadInt%"
+      val emsLength = textLengthCalculator.getTextLength(loadStr)
+      val displayLoad = loadInt != 100 && emsLength <= rect.width
+      if (displayLoad) arrayOf(loadLabel.createLabel(loadStr, rect.width)) else arrayOfNulls(0)
```
Comment (dbarashev) [inline biz.ganttproject.core/src/biz/ganttproject/core/chart/scene/CapacityHeatmapSceneBuilder.kt:126]:
```suggestion
      if (displayLoad) arrayOf(loadLabel.createLabel(loadStr, rect.width)) else emptyArray()
```

File: ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java [hunk 2]
```diff
@@ -68,200 +49,73 @@ public ResourceLoadRenderer(ChartModelResource model, ResourceChart resourceChar
...
-    public String toString() {
-      return Float.toString(load);
-    }
+    return resources;
   }
 
-  private int calculateGap(HumanResource resource) {
-    return resource.getAssignments().length * getConfig().getRowHeight();
+  private long toTs(Date date) {
+    return date.toInstant().toEpochMilli();
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java:119]: Date.getTime() does the same

File: ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java [hunk 3]
```diff
@@ -68,200 +49,73 @@ public ResourceLoadRenderer(ChartModelResource model, ResourceChart resourceChar
...
-
-    public float getLoad() {
-      return load;
+  private List<CapacityHeatmapSceneBuilder.Resource> getResources() {
+    HumanResource[] humanResources = ((ChartModelResource) getChartModel()).getVisibleResources();
+    List<CapacityHeatmapSceneBuilder.Resource> resources = new ArrayList<>();
+
+    for (HumanResource humanResource : humanResources) {
+      List<CapacityHeatmapSceneBuilder.Load> loads = new ArrayList<>();
+      List<Load> tasksLoads = humanResource.getLoadDistribution().getTasksLoads();
```
Comment (dbarashev) [inline ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java:101]: Apparently LoadDistribution will do the same stuff which we do in biz.ganttproject.core.chart.scene.CapacityHeatmapSceneBuilder#calcLoadDistribution ? Can you add TODO to remove this work duplication?

Comment (DmKazakov) [reply ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java:101]: True, but we can't use LoadDistribution in GP cloud. Should I remove that duplication in LoadDistribution then?

Comment (dbarashev) [reply ganttproject/src/net/sourceforge/ganttproject/chart/ResourceLoadRenderer.java:101]: Yes, if it is no more useful for GP desktop purposes.

**Review by dbarashev — APPROVED** [review]

**Pull request merged**